hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -683,6 +683,8 @@ public QueryModel createQueryModel(InputSplit inputSplit, TaskAttemptContext tas
     }
     if (indexFilter != null) {
       checkAndAddImplicitExpression(indexFilter.getExpression(), inputSplit);
+      // process filter expression after adding implicit expression
+      indexFilter.processFilterExpression();
```

Comment:
in `Querymodelbuilder.build()` we call `processFilterExpression` with check of columndrift, so here if we call process, then in case of no column drift , two times will be called, so please refactor to call this only in case of columndrift true

Reply:
done